cmd/net.go

```diff
@@ -37,42 +37,77 @@ func mustSplitHostPort(hostPort string) (host, port string) {
 	return host, port
 }
 
+// [The logic to sort by last octet is implemented to
+// prefer CIDRs with higher octects, this in-turn skips the
+// localhost/loopback address to be not preferred as the
+// first ip on the list. Subsequently this list helps us print
+// a user friendly message with appropriate values].
+
+// byLastOctetValue implements sort.Interface used in sorting a list
+// of ip address by their last octet value.
+type byLastOctetValue []net.IP
+
+func (n byLastOctetValue) Len() int      { return len(n) }
+func (n byLastOctetValue) Swap(i, j int) { n[i], n[j] = n[j], n[i] }
+func (n byLastOctetValue) Less(i, j int) bool {
+	return []byte(n[i].To4())[3] > []byte(n[j].To4())[3]
+}
+
 // mustGetLocalIP4 returns IPv4 addresses of local host.  It panics on error.
-func mustGetLocalIP4() (ipList set.StringSet) {
-	ipList = set.NewStringSet()
+func mustGetLocalIP4() (ipList []string) {
```

Comment:
We need to sort and save it in the form we are going to use.  That is the only purpose of APIEndpoints() .  why would i parse again it into a net.IP form and then print it.